src/Workspaces/Core/Portable/SymbolKey/SymbolKey.AnonymousFunctionOrDelegateSymbolKey.cs

```diff
@@ -18,7 +18,10 @@ private static class AnonymousFunctionOrDelegateSymbolKey
         {
             public static void Create(ISymbol symbol, SymbolKeyWriter visitor)
             {
-                Debug.Assert(symbol.IsAnonymousDelegateType() || symbol.IsAnonymousFunction());
+                Debug.Assert(
```

Comment:
File/type needs to be renamed to reflect it can handle LocalFunctions.  Though i find the use of this type weird.  anonymous-functions/delegates are rather strange beasts.  I would think you could have used "BodyLevelSymbolKey" and just had this work.